x-pack/packages/kbn-langchain/server/utils/gemini.ts

```diff
@@ -5,8 +5,38 @@
  * 2.0.
  */
 
+import { Logger } from '@kbn/core/server';
+import { Readable } from 'stream';
 import { StreamParser } from './types';
 
+export const parseGeminiStreamAsAsyncIterator = async function* (
+  stream: Readable,
+  logger: Logger,
+  abortSignal?: AbortSignal
+) {
+  if (abortSignal) {
+    abortSignal.addEventListener('abort', () => {
+      stream.destroy();
+    });
+  }
+  try {
+    for await (const chunk of stream) {
+      const decoded = chunk.toString();
+      const parsed = parseGeminiResponse(decoded);
+      // Split the parsed string into chunks of 5 characters
+      for (let i = 0; i < parsed.length; i += 5) {
+        yield parsed.substring(i, i + 5);
+      }
```

Comment:
Any reason for splitting at 5 characters, or is that just what ends up working out best for responsiveness/visually?

Reply:
1 character is too slow, 5 looks nice. without it, its like 3 big chunks and it looks not like a stream